capellambse_context_diagrams/collectors/default.py

```diff
@@ -75,6 +84,20 @@ def collector(
             else:
                 global_boxes[i.uuid] = box
 
+        if diagram.display_parent_relation:
+            if i == diagram.target.parent:
+                _move_edge_to_local_edges(
+                    box, connections, local_ports, diagram, data
+                )
+            elif i.parent == diagram.target.parent:
+                parent_box = global_boxes[i.parent.uuid]
+                parent_box.setdefault("children", []).append(
+                    global_boxes.pop(i.uuid)
+                )
+                _move_edge_to_local_edges(
+                    parent_box, connections, local_ports, diagram, data
+                )
+
         stack_heights[side] += makers.NEIGHBOR_VMARGIN + height
 
```

Comment:
```suggestion
                edges_to_remove += _move_edge_to_local_edges(
                    box, connections, local_ports, diagram, data
                )
            elif i.parent == diagram.target.parent:
                parent_box = global_boxes[i.parent.uuid]
                parent_box.setdefault("children", []).append(
                    global_boxes.pop(i.uuid)
                )
                edges_to_remove += _move_edge_to_local_edges(
                    parent_box, connections, local_ports, diagram, data
                )

        stack_heights[side] += makers.NEIGHBOR_VMARGIN + height
    
    data["edges"] = [
        e for e in data["edges"] if e["id"] not in edges_to_remove
    ]
```

O(N*E) vs O(E) I think?

Reply:
Still need to call `_move_edge_to_local_edges` inside the for-loop. With ur suggestion we'd have O((N + 1)*E).